daemonset: added WithHostNetwork and WithVolume functions

ajaggapa opened the pull request:
Added functions to update the daemonset builder with Hostnetwork and configmap volume mount.

kononovn commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+		return builder
+	}
+
+	glog.V(100).Infof("Enabling hostnetwork to daemonset %s in namespace %s",
```
Please change log message to: "Applying HostNetwork flagdaemonset %s in namespace %s"

ajaggapa replied:
updated as suggested

kononovn commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+	return builder
+}
+
+// WithConfigMapVolume attaches given volume to all pod's containers.
```
Please update func comment to something like:
// WithConfigMapVolume attaches given volume to all pod's containers under daemonset.

ajaggapa replied:
updated as suggested

kononovn commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+		return builder
+	}
+
+	glog.V(100).Infof("Configuring volume %s for all pod's: %s containers. MountPath %s",
```
Please adjust comment to daemonset. It's copy/paste from pod package.

ajaggapa replied:
updated as suggested

kononovn commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies HostNetwork to pod's definition.
```
It's not pod definition. It's daemonset definition.

ajaggapa replied:
updated as suggested

kononovn reviewed (changes requested):
Please fix few comments above.

kononovn reviewed (changes requested):
Please change PR headline and commit message to format:
package name: what was the change
Example:
sriov: added WithHostNetwork,WithConfigMapVolume  funcs

ajaggapa commented:
> Please change PR headline and commit message to format: package name: what was the change Example: sriov: added WithHostNetwork,WithConfigMapVolume funcs

Updated

kononovn reviewed (dismissed):
/lgtm

cdvultur reviewed (dismissed):
lgtm

achuzhoy commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying hostnetwork flag to daemonset %s in namespace %s",
```
Applying or Enabling?

ajaggapa replied:
This gets tricky on whats makes more sense.
Started off with "Enabling", had to modify to "Applying" after @kononovn review.

trewest commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,71 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+}
+
+// WithConfigMapVolume attaches given volume to all pod's containers under daemonset.
+func (builder *Builder) WithConfigMapVolume(volumeName, mountPath string) *Builder {
```
Can you align this method with the WithVolume deployment method?
https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L301C25-L301C35

trewest replied:
For context, volumeMounts should be added through ContainerBuilder - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/pod/container.go#L353

ajaggapa replied:
I was aligning with "WithLocalVolume" method of pod pkg. But yes, it is more structured to have separate methods for Volume and VolumeMount (container builder). 
I have made changes accordingly. PTAL.

trewest replied:
Understood, thanks. Will make an issue for refactoring that one

trewest reviewed (changes requested):
One comment

trewest commented on pkg/daemonset/daemonset.go:
```diff
@@ -138,6 +138,51 @@ func (builder *Builder) WithNodeSelector(selector map[string]string) *Builder {
+}
+
+// WithConfigMapVolume defines Volume of daemonset under PodTemplateSpec with given ConfigMap.
+func (builder *Builder) WithConfigMapVolume(cmName string) *Builder {
```
Hey sorry for the confusion but the idea is to just make this a generic WithVolume function. We don't want to limit it to only configmaps and creating a new mutation for each volume source type would be a lot of work and difficult to manage. For the most part it is just a copy-paste of with WithVolume method from the deployment pkg

trewest reviewed (approved):
/lgtm
